Profiler/CProfilerCallback.h

```diff
@@ -1,157 +1,184 @@
+	/**
+	 * Keeps track of inlined methods.
+	 * We use the vector to uniquely store the information about inlined methods.
+	 */
+	std::vector<FunctionInfo> inlinedMethods;
+
+	/** Smart pointer to the .NET framework profiler info. */
+	CComQIPtr<ICorProfilerInfo8> profilerInfo;
+
+	/** The log to write all results and messages to. */
+	TraceLog traceLog;
+
+	/** The log to write attach and detatch events to */
+	AttachLog attachLog;
+
+	/** Inter-process connection for TIA communication. null if not in TIA mode. */
+	Ipc* ipc = NULL;
+
+	/** Callback that is being called when a testcase starts. */
+	void onTestStart(std::string testName);
+
+	/** Callback that is being called when a testcase ends. */
+	void onTestEnd(std::string result = "", std::string duration = "");
+
+	/**
+	 * Keeps track of called methods.
+	 * We use the set to efficiently determine if we already noticed an called method.
+	 */
+	function_id_set calledMethodIds;
+
+	/**
+	 * Keeps track of called methods.
+	 * We use the vector to uniquely store the information about called methods.
+	 */
+	std::vector<FunctionInfo> calledMethods;
+private:
+
+	/**
+	* Returns the event mask which tells the CLR which callbacks the profiler wants to subscribe
+	* to. We enable JIT compilation and assembly loads for coverage profiling. In
+	* addition if light mode is disabled, EnterLeave hooks are enabled to force re-jitting of pre-jitted
+	* code, in order to make coverage information independent of pre-jitted code.
+	*/
+	DWORD getEventMask();
+
+	/**
+	* Defines whether the given function should trigger a callback everytime it is executed.
+	*
+	* We do not register a single function callback in order to not affect
+	* performance. In effect, we disable this feature here. It has been tested via
+	* a performance benchmark, that this implementation does not impact call
+	* performance.
+	*
+	* For coverage profiling, we do not need this callback. However, the event is
+	* enabled in the event mask in order to force JIT-events for each first call to
+	* a function, independent of whether a pre-jitted version exists.)
+	*/
+	static UINT_PTR _stdcall functionMapper(FunctionID functionId, BOOL* pbHookFunction) throw(...);
+	void adjustEventMask();
+
+	/** Dumps all environment variables to the log file. */
+	void dumpEnvironment();
+
+	void initializeConfig();
+
+	/** Returns a proxy for the upload daemon process */
+	UploadDaemon createDaemon();
+
+	/** Create method info object for a function id. */
+	HRESULT getFunctionInfo(FunctionID functionID, FunctionInfo* info);
+
+	/**  Store assembly counter for id. */
+	int registerAssembly(AssemblyID assemblyId);
+
+	/** Stores the assmebly name, path and metadata in the passed variables.*/
+	void getAssemblyInfo(AssemblyID assemblyId, WCHAR* assemblyName, WCHAR* assemblyPath, ASSEMBLYMETADATA* moduleId);
+
+	/** Triggers eagerly writing of function infos to log. */
+	void recordFunctionInfo(std::vector<FunctionInfo>* list, FunctionID calleeId);
```

Comment:
Prefer using references for passing vectors, to have more type safety.
Consider if parameters are in (only read), out (return value), or in/out (read/write/return), see [cpp core guidelines on parameter passing](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f15-prefer-simple-and-conventional-ways-of-passing-information)
```suggestion
	void recordFunctionInfo(std::vector<FunctionInfo>& list, FunctionID calleeId);
```